docs/current_docs/api/module-structure.mdx

```diff
@@ -461,5 +530,28 @@ composer install
 
 This means that so long as the project has a `composer.json` file, it can be used as a library.
 
+</TabItem>
+<TabItem value="Java">
+Dagger modules in Java are built as JAR file, using Maven. The runtime container builds the module code with:
+
+```shell
+# executed by the runtime build container
+mvn clean package
+```
+
+The generated JAR file is then executed by the runtime container with:
+
+```shell
+# executed by the runtime run container
+java -jar module.jar
+```
+
+:::important
+This means you have to keep the `pom.xml` file and especially the `maven-compiler-plugin` and the `maven-shade-plugin` configurations.
```

Comment:
This sentence is not very clear. Where should the user keep these files?

Reply:
The `pom.xml` is at the root of the module source code. This is the very standard place for it. The configurations are blocks inside the `pom.xml`, again very standard way.